marklogic-data-hub-central/ui/e2e/cypress/integration/modeling/reader.spec.tsx

```diff
@@ -0,0 +1,93 @@
+/// <reference types="cypress"/>
+
+import modelPage from '../../support/pages/model';
+import {
+  entityTypeModal,
+  entityTypeTable,
+  propertyModal,
+  propertyTable,
+} from '../../support/components/model/index';
+import { confirmationModal, toolbar, tiles } from '../../support/components/common/index';
+import { Application } from '../../support/application.config';
+import browsePage from '../../support/pages/browse'
+import 'cypress-wait-until';
+
+describe('Entity Modeling: Reader Role', () => {
+
+  //login with valid account
+  beforeEach(() => {
+    cy.visit('/');
+    cy.contains(Application.title);
+    console.log(Cypress.env('mlHost'));
+    cy.loginAsTestUserWithRoles("hub-central-entity-model-reader").withRequest()
+    cy.waitUntil(() => toolbar.getModelToolbarIcon()).click();
+    entityTypeTable.waitForTableToLoad();
+  });
+
+  after(() => {
+      //resetting the test user back to only have 'hub-central-user' role
+      cy.resetTestUser();
+  });
+
+  it('can navigate by clicking instance count and last processed, can not create, edit, or delete entity models', () => {
+    cy.waitUntil(() => entityTypeTable.getEntityLastProcessed('Person')).click();
+    cy.url().should('include', '/tiles/explore');
+    tiles.getExploreTile().should('exist');
+    cy.waitUntil(() => browsePage.getSelectedEntity()).should('eq', 'Person');
+    browsePage.getClearAllButton().should('exist');
+    
+    toolbar.getModelToolbarIcon().click();
+    cy.url().should('include', '/tiles/model');
+    tiles.getModelTile().should('exist');
+
+    cy.waitUntil(() => entityTypeTable.getEntityInstanceCount('Order')).click();
+    cy.url().should('include', '/tiles/explore');
+    tiles.getExploreTile().should('exist');
+    cy.waitUntil(() => browsePage.getSelectedEntity().should('eq', 'Order'));
+    browsePage.getClearAllButton().should('not.exist');
+
+    cy.go('back');
+    cy.url().should('include', '/tiles/model');
+    tiles.getModelTile().should('exist');
+
+    modelPage.getAddEntityButton().click({ force: true });
```

Comment:
wondering why you had to use force: true option.

Reply:
The test will fail if I don't because they aren't actionable items. Even though the button is disabled, I'm forcing clicks to see if the functionality is there. For edit Entity/Property names in the table, model-reader role displays text instead of links (with the same selector). So for reader tests, I'm clicking the same selectors as writer role, but I don't expect any edit/ add modals to show up.

Running the without force: true
![Entity Modeling Reader Role -- can navigate by clicking instance count and last processed, can not create, edit, or delete entity models (failed)](https://user-images.githubusercontent.com/32427979/89556783-bea1e980-d7c6-11ea-808d-0d76765e5155.png)

Reply:
Okay. interesting way to test a reader role. So you are forcing a click action to verify that the user is able to read?

Reply:
I'm testing that the reader role doesn't have any of the writer role privileges in the UI. If you add the "model writer" role to the reader test, the test will fail.